### PR TITLE
Code refactoring and bug fixing

### PR DESCRIPTION
**Pull request by GRaykovski**

### CODE REVIEW

File: thread_pool.h [hunk 1]
```diff
@@ -32,13 +33,13 @@ class ThreadPool {
 
 private:
     RejectedJobPolicy* rejected_job_policy;
-    unsigned num_active_threads = 0;
+    unsigned num_active_threads;
```
Comment (radito3) [inline thread_pool.h:36]: Why is this removed?

File: thread_pool.h [hunk 2]
```diff
@@ -55,7 +56,7 @@ class ThreadPool {
         }
 
         for(int i = 0 ; i < num_active_threads ; i++){
-            task_queue.emplace_back([]() {});
+            task_queue.emplace([]() {});
```
Comment (radito3) [inline thread_pool.h:59]: Why is this changed?

**Review by radito3 — CHANGES_REQUESTED** [review]
Revert the search/lobbby id change to string. It should be a number

File: thread_pool.h (hunk 2 above)
Comment (GRaykovski) [reply thread_pool.h:59]: There is no queue.emplace_back method. Should I change task_queue to a vector ?

Comment (radito3) [reply thread_pool.h:59]: Should you? What is the difference between a queue and a vector? Do we need a vector here?

File: thread_pool.h (hunk 1 above)
Comment (GRaykovski) [reply thread_pool.h:36]: > Revert the search/lobbby id change to string. It should be a number

Why ? If we keep it as an int that will only add redundant code every time we try to pass this variable in a response. This way we only parse it once when the search is created and we don't have to worry about anything later on.

**Comment by GRaykovski** [comment]
> Revert the search/lobbby id change to string. It should be a number

Why ? If we keep it as an int that will only add redundant code every time we try to pass this variable in a response. This way we only parse it once when the search is created and we don't have to worry about anything later on.

File: thread_pool.h (hunk 1 above)
Comment (GRaykovski) [reply thread_pool.h:36]: Either way we set this variable every time we initialize the Thread Pool so i do no think this has any impact whatsoever.

**Comment by radito3** [comment]
I disagree. Where would the redundancy be? We only pass it to the response body once. Having it as an integer makes it clearer what type of data it is. And passing it to and returning it from methods as a string is less memory efficient than as an int.

File: thread_pool.h (hunk 2 above)
Comment (GRaykovski) [reply thread_pool.h:59]: No we do not need a vector.
This was changed for 2 reasons : 
First as I already said there is no queue.emplace_back() method
And second this method does the exact same thing as the emplace_back method for vectors

**Review by radito3 — APPROVED** [review]
